File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-fortran-skills/tsvc_2_s235__submitted.f90
subroutine tsvc_2_s235_fp64(a, aa, b, bb, c, LEN_2D) bind(C, name='tsvc_2_s235_fp64')
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: LEN_2D
  real(c_double), intent(inout) :: a(LEN_2D)
  real(c_double), intent(inout) :: aa(LEN_2D, LEN_2D)
  real(c_double), intent(in) :: b(LEN_2D)
  real(c_double), intent(in) :: bb(LEN_2D, LEN_2D)
  real(c_double), intent(in) :: c(LEN_2D)
  integer(c_int64_t) :: i, j, n

  n = LEN_2D

  !$omp parallel proc_bind(spread)

  !$omp do simd schedule(static)
  do i = 1, n
     a(i) = a(i) + b(i) * c(i)
  end do
  !$omp end do simd nowait

  do j = 2, n
     !$omp do simd schedule(static)
     do i = 1, n
        aa(i, j) = aa(i, j - 1) + bb(i, j) * a(i)
     end do
     !$omp end do simd nowait
  end do

  !$omp end parallel

end subroutine tsvc_2_s235_fp64

File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-c/tsvc_2_s235__submitted.c
#include <stdint.h>

void tsvc_2_s235_fp64(double *restrict a, double *restrict aa,
                      const double *restrict b, const double *restrict bb,
                      const double *restrict c, int64_t LEN_2D,
                      uint8_t *restrict workspace, int64_t workspace_bytes)
{
    #pragma omp parallel
    {
        #pragma omp for simd
        for (int64_t i = 0; i < LEN_2D; i++) {
            a[i] = a[i] + b[i] * c[i];
        }

        for (int64_t j = 1; j < LEN_2D; j++) {
            #pragma omp for simd
            for (int64_t i = 0; i < LEN_2D; i++) {
                aa[j * LEN_2D + i] = aa[(j - 1) * LEN_2D + i] + bb[j * LEN_2D + i] * a[i];
            }
        }
    }
}
